python/cudf/cudf/core/column/lists.py

```diff
@@ -451,3 +457,56 @@ def sort_values(
             sort_lists(self._column, ascending, na_position),
             retain_index=not ignore_index,
         )
+
+    def ravel(self) -> ParentType:
```

Comment:
Could use some help here with naming/docstring here.

Unlike `np.ravel`, this function only removes _one_ level of nesting from each row. What's a better name for that?

Reply:
> What's a better name for that?

`unpack`, `unbox` ? (along with a parameter `n` for how many levels of nesting to be removed? - but maybe this is something we can do in a future PR)

Reply:
I like `unbox`, and I also like the suggestion for an `n` parameter

Reply:
What about `flatten`? That's what Spark uses: https://spark.apache.org/docs/latest/api/sql/index.html#flatten

Reply:
Sounds good to me.

Reply:
Do we still want the `n` parameter with `flatten`?

Reply:
Looks like Spark `flatten` only flattens one level, whereas numpy `flatten` does similar to `ravel` where it flattens down to 1d: https://numpy.org/doc/stable/reference/generated/numpy.ndarray.flatten.html#numpy.ndarray.flatten

Any thoughts on what makes the most sense for us?

Reply:
I went with `lists.concat()` as that's really what we're doing (concatenating the inner lists). If we only have a single inner list, nesting is removed:

```python
In [15]: s
Out[15]:
0    [[1, 2], [3, 4]]
dtype: list

In [16]: s.list.concat()
Out[16]:
0    [1, 2, 3, 4]
dtype: list
```


```python
In [18]: s
Out[18]:
0    [[1, 2]]
dtype: list

In [19]: s.list.concat()
Out[19]:
0    [1, 2]
dtype: list
```